diff --git a/conan/tools/build/cppstd.py b/conan/tools/build/cppstd.py
index 1bea2f41d..28d4a7156 100644
--- a/conan/tools/build/cppstd.py
+++ b/conan/tools/build/cppstd.py
@@ -102,6 +102,7 @@ def supported_cppstd(conanfile, compiler=None, compiler_version=None):
             "gcc": _gcc_supported_cppstd,
             "msvc": _msvc_supported_cppstd,
             "clang": _clang_supported_cppstd,
+            "ibm-clang": _ibm_clang_supported_cppstd,
             "mcst-lcc": _mcst_lcc_supported_cppstd,
             "qcc": _qcc_supported_cppstd,
             }.get(compiler)
@@ -174,6 +175,16 @@ def _apple_clang_supported_cppstd(version):
             "26", "gnu26"]
 
 
+def _ibm_clang_supported_cppstd(version):
+    """
+    ["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"]
+    """
+    if version < "1.1":
+        return []
+
+    return ["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"]
+
+
 def _gcc_supported_cppstd(version):
     """
     ["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17", "20", "gnu20", "23", "gnu23"]
