diff --git a/conan/tools/build/cstd.py b/conan/tools/build/cstd.py
index 2d69eaba9..bfe13d7f5 100644
--- a/conan/tools/build/cstd.py
+++ b/conan/tools/build/cstd.py
@@ -94,6 +94,7 @@ def supported_cstd(conanfile, compiler=None, compiler_version=None):
             "gcc": _gcc_supported_cstd,
             "msvc": _msvc_supported_cstd,
             "clang": _clang_supported_cstd,
+            "ibm-clang": _ibm_clang_supported_cstd,
             }.get(compiler)
     if func:
         return func(Version(compiler_version))
@@ -174,3 +175,9 @@ def _clang_supported_cstd(version):
     if version < "18":
         return ["99", "gnu99", "11", "gnu11", "17", "gnu17"]
     return ["99", "gnu99", "11", "gnu11", "17", "gnu17", "23", "gnu23"]
+
+
+def _ibm_clang_supported_cstd(version):
+    if version < "1.1":
+        return []
+    return ["99", "gnu99", "11", "gnu11", "17", "gnu17"]
