diff --git a/conan/internal/default_settings.py b/conan/internal/default_settings.py
index 0c6f0db6b..7d7c88192 100644
--- a/conan/internal/default_settings.py
+++ b/conan/internal/default_settings.py
@@ -87,6 +87,7 @@ os:
     baremetal:
     VxWorks:
         version: ["7"]
+    z/OS:
 arch: [x86, x86_64, ppc32be, ppc32, ppc64le, ppc64,
        armv4, armv4i, armv5el, armv5hf, armv6, armv7, armv7hf, armv7s, armv7k, armv8, armv8_32, armv8.3, arm64ec,
        sparc, sparcv9,
@@ -141,6 +142,14 @@ compiler:
         libcxx: [libstdc++, libc++]
         cppstd: [null, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20, 23, gnu23, 26, gnu26]
         cstd: [null, 99, gnu99, 11, gnu11, 17, gnu17, 23, gnu23]
+    ibm-clang:
+        version: ["1.1", "2.1"]
+        libcxx: [null, libc++, c++_shared, c++_static]
+        cppstd: [null, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17]
+        runtime: [null, static, dynamic]
+        runtime_type: [null, Debug, Release]
+        runtime_version: [null, v140, v141, v142, v143, v144]
+        cstd: [null, 99, gnu99, 11, gnu11, 17, gnu17]
     intel-cc:
         version: ["2021.1", "2021.2", "2021.3", "2021.4", "2022.1", "2022.2",
                   "2022.3", "2023.0", "2023.1", "2023.2", "2024.0",]
